### ISSUE TITLE
git is required for installation

### ISSUE THREAD

**Issue by alexandrupaler** [ngnrsaa/qflex#278]
If somebody downloads the code in a zip file and does not have git installed, the installer does not work. The `install.md` document should mention that git is prerequisite.

autoconf script should also check this, because git is necessary for preparing the compilation of the pypi package.

**Comment by 95-martin-orion** [comment]
> autoconf script should also check this, because git is necessary for preparing the compilation of the pypi package.

Agreed. Before this I think we assumed that the user had git installed (how else would they have cloned the repo?) but with the PyPI package we need to double-check.

**Comment by alexandrupaler** [comment]
> how else would they have cloned the repo?

I used to download zip files of the repo from github

**Comment by s-mandra** [comment]
`git` is not strictly required to run `qflex`, but some scripts like `check_format.sh` require it. I'm ok to either remove the check from `configure.ac` add it to the list of required software.

**Comment by alexandrupaler** [comment]
`git` is required to compile qflexcirq because the code includes a submodule. compilation is lamost always required each time the user downloads qflexcirq from pypi. for this reason there I had to add two lines that invoke git in the `setup.py` file.

**Comment by s-mandra** [comment]
> `git` is required to compile qflexcirq because the code includes a submodule. compilation is almost always required each time the user downloads qflexcirq from pypi. for this reason there I had to add two lines that invoke git in the `setup.py` file.

Got it. I didn't know about this. Let's add it to `install.md` (`configure` already checks for its installation).

**Comment by alexandrupaler** [comment]
There is also a `docopt` in `requirements.txt`. Should we treat `docopt` like `pybind` and not use submodules but pypi package instead? This could also speed-up compilation of python wheels.
